google/cloud/datastore/query.py

```diff
@@ -804,46 +813,69 @@ def _next_page(self):
             "project_id": self._query.project,
             "partition_id": partition_id,
             "read_options": read_options,
-            "query": query_pb,
+            "query": self._build_protobuf(),
         }
+        if self._query._explain_options:
+            request["explain_options"] = self._query._explain_options._to_dict()
 
         helpers.set_database_id_to_request(request, self.client.database)
 
-        response_pb = self.client._datastore_api.run_query(
-            request=request,
-            **kwargs,
-        )
+        response_pb = None
 
-        while (
+        while response_pb is None or (
```

Comment:
The new while loop seems mostly the same as the previous one. Just for my education,  could you explain how they are different and why we need to make the change here?

Reply:
Adding `if response_pb is None` is essentially changing it from a `while` loop to a `do while` loop; Previously the implementation would doa. single call before the loop, and then repeat a bunch of code to repeat the call inside the loop as well. I wanted to reduce the duplicated code, and move everything inside the loop

The other change was to get rid of the `query_pb` variable, and directly access the request inside the query instead, to keep all the state in one place